test/jdk/java/foreign/StdLibTest.java

```diff
@@ -305,8 +305,10 @@ int rand() throws Throwable {
         }
 
         int printf(String format, List<PrintfArg> args) throws Throwable {
+            byte[] nativeEncodingBytes = format.getBytes(System.getProperty("native.encoding"));
```

Comment:
Native encoding should be applied not just to the format argument, but all string arguments to `printf()` (i.e. `PrintfArg.STRING` must do the same). Both should probably use `MemorySegment.allocateFrom(String, Charset)`.

This file should also get an IBM copyright notice.

Reply:
Yes, I did think of it and trying to make the change but the issue is that PrintfArg.STRING  is using enum  and both MemorySegment.allocateFrom(String, Charset) or getting Bytes from String with native encoding throws UnsupportedEncodingException, which I am trying to figure out how to handle when enum constructor is called for `STRING`.  Simple try catch  for lambda is not working. 

I have made the copyright changes.

Reply:
Checking if the static method can be called that handles the exception, and seems like it is working, Let me make the change.